lib/Driver/ToolChains.cpp

```diff
@@ -1101,33 +1101,49 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
 
   ArgStringList Arguments;
 
+  // Explicitly pass the target to the linker
+  Arguments.push_back(context.Args.MakeArgString("--target=" + getTriple().str()));
```

Comment:
Nitpick: Let's do this as two strings: `--target` and the triple string. That's slightly less allocation of memory in constructing the argument list.